utils/logging/log.go

```diff
@@ -294,73 +296,91 @@ func (l *Log) SetLogLevel(lvl Level) {
 	l.config.LogLevel = lvl
 }
 
-// SetDisplayLevel ...
+// SetDisplayLevel implements the Logger interface
 func (l *Log) SetDisplayLevel(lvl Level) {
 	l.configLock.Lock()
 	defer l.configLock.Unlock()
 
 	l.config.DisplayLevel = lvl
 }
 
-// SetPrefix ...
+// SetPrefix implements the Logger interface
 func (l *Log) SetPrefix(prefix string) {
 	l.configLock.Lock()
 	defer l.configLock.Unlock()
 
 	l.config.MsgPrefix = prefix
 }
 
-// SetLoggingEnabled ...
+// SetLoggingEnabled implements the Logger interface
 func (l *Log) SetLoggingEnabled(enabled bool) {
 	l.configLock.Lock()
 	defer l.configLock.Unlock()
 
 	l.config.DisableLogging = !enabled
 }
 
-// SetDisplayingEnabled ...
+// SetDisplayingEnabled implements the Logger interface
 func (l *Log) SetDisplayingEnabled(enabled bool) {
 	l.configLock.Lock()
 	defer l.configLock.Unlock()
 
 	l.config.DisableDisplaying = !enabled
 }
 
-// SetContextualDisplayingEnabled ...
+// SetContextualDisplayingEnabled implements the Logger interface
 func (l *Log) SetContextualDisplayingEnabled(enabled bool) {
 	l.configLock.Lock()
 	defer l.configLock.Unlock()
 
 	l.config.DisableContextualDisplaying = !enabled
 }
 
+// fileWriter implements the RotatingWriter interface
 type fileWriter struct {
 	writer *bufio.Writer
 	file   *os.File
 
-	config    Config
-	fileIndex int
+	config Config
 }
 
+// Flush implements the RotatingWriter interface
 func (fw *fileWriter) Flush() error {
 	return fw.writer.Flush()
 }
 
+// Write implements the RotatingWriter interface
 func (fw *fileWriter) Write(b []byte) (int, error) {
 	return fw.writer.Write(b)
 }
 
+// WriteString implements the RotatingWriter interface
 func (fw *fileWriter) WriteString(s string) (int, error) {
 	return fw.writer.WriteString(s)
 }
 
+// Close implements the RotatingWriter interface
 func (fw *fileWriter) Close() error {
 	return fw.file.Close()
 }
 
+// Rotate implements the RotatingWriter interface
 func (fw *fileWriter) Rotate() error {
-	fw.fileIndex = (fw.fileIndex + 1) % fw.config.RotationSize
-	writer, file, err := fw.create(fw.fileIndex)
+	for i := fw.config.RotationSize - 1; i >= 0; i-- {
+		sourceFilename := filepath.Join(fw.config.Directory, fmt.Sprintf("%s.log.%d", fw.config.LoggerName, i))
+		destFilename := filepath.Join(fw.config.Directory, fmt.Sprintf("%s.log.%d", fw.config.LoggerName, i+1))
+		if _, err := os.Stat(sourceFilename); !os.IsNotExist(err) {
```

Comment:
https://stackoverflow.com/a/12518877/169252

Reply:
Thank you. Followed the most recent comment in the answer which suggests using `errors.Is()` as it now seems to be the preferred way to check for file existence as per  https://github.com/golang/go/blob/5ff7ec98b7727b3641df25200345b1aa50b6ff35/src/os/error.go#L90-L91